python/plugins/processing/tools/vector.py

```diff
@@ -69,6 +70,13 @@
     QVariant.Bool: "BOOLEAN"
 }
 
+TYPE_MAP_SPATIALITE_LAYER = {
+    QVariant.String: "VARCHAR",
+    QVariant.Double: "REAL",
+    QVariant.Int: "INTEGER",
+    QVariant.Bool: "INTEGER"
```

Comment:
Boolean does not exist in SQLite.
In the documentation http://www.sqlite.org/datatype3.html you'll find that Boolean values are stored as integers 0 (false) and 1 (true).